### PR TITLE
[Grade] Backend of the feature

### PR DESCRIPTION
**Pull request by apomalyn**
Alright! There is the first part of the feature. 
Contains:
- Get the list of courses
- Get the summary of a course
- When a course doesn't have a grade, get the course summary

Acceptation criteria:
- [x] Technical documentation
- [x] Tests

### CODE REVIEW

File: android/settings.gradle [hunk 1]
```diff
@@ -13,3 +13,16 @@ localPropertiesFile.withReader("UTF-8") { reader -> properties.load(reader) }
 def flutterSdkPath = properties.getProperty("flutter.sdk")
 assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
 apply from: "$flutterSdkPath/packages/flutter_tools/gradle/app_plugin_loader.gradle"
+def flutterProjectRoot = rootProject.projectDir.parentFile.toPath()
```
Comment (AlexandreMarkus) [inline android/settings.gradle:16]: what's the usage?

**Review by AlexandreMarkus — COMMENTED** [review]
still need to test it on my machine.

UI was not reviewed

File: android/settings.gradle (hunk 1 above)
Comment (apomalyn) [reply android/settings.gradle:16]: Good question, I didn't saw the change. Will remove it!

Comment (AlexandreMarkus) [reply android/settings.gradle:16]: after this, seems good to accept

File: android/settings.gradle [hunk 2]
```diff
@@ -8,8 +8,3 @@ def localPropertiesFile = new File(rootProject.projectDir, "local.properties")
 def properties = new Properties()
 
 assert localPropertiesFile.exists()
-localPropertiesFile.withReader("UTF-8") { reader -> properties.load(reader) }
```
Comment (AlexandreMarkus) [inline android/settings.gradle:11]: we need to keep these 5 lines to be able to build the app.
You removed too much lines ;)

**Review by AlexandreMarkus — CHANGES_REQUESTED** [review]

File: android/settings.gradle (hunk 2 above)
Comment (AlexandreMarkus) [reply android/settings.gradle:11]: localPropertiesFile.withReader("UTF-8") { reader -> properties.load(reader) }

def flutterSdkPath = properties.getProperty("flutter.sdk")
assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
apply from: "$flutterSdkPath/packages/flutter_tools/gradle/app_plugin_loader.gradle"

Comment (apomalyn) [reply android/settings.gradle]: Oups XD

**Review by AlexandreMarkus — APPROVED** [review]

**Pull request merged**